server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -7546,6 +7546,89 @@ public void testGetSegmentInfosSnapshot() throws IOException {
         engine.close();
     }
 
+    // This test performs indexing ops and verifies engine.getSegmentInfosSnapshot() returns SegmentInfos having highest
+    // generation
+    public void testGetSegmentInfosSnapshotWithIndexing() throws IOException {
```

Comment:
given you have tests for both cases where the infos returned is from disk and from memory, I don't think this test provides much additional value.

Reply:
Removing this test.